Sources/Workspace/ManagedDependency.swift

```diff
@@ -14,195 +14,192 @@ import SourceControl
-        }
-        return nil
-    }
+        /// The checked out path of the dependency on disk, relative to the workspace checkouts path.
+        public let subpath: RelativePath
 
-    /// The checked out path of the dependency on disk, relative to the workspace checkouts path.
-    public let subpath: RelativePath
-
-    /// A dependency which in editable state is based on a dependency from
-    /// which it edited from.
-    ///
-    /// This information is useful so it can be restored when users
-    /// unedit a package.
-    public internal(set) var basedOn: ManagedDependency?
-
-    public var packageIdentity: PackageIdentity {
-        self.packageRef.identity
-    }
+        public var packageIdentity: PackageIdentity {
+            self.packageRef.identity
+        }
 
-    public init(
-        packageRef: PackageReference,
-        subpath: RelativePath,
-        checkoutState: CheckoutState
-    ) {
-        self.packageRef = packageRef
-        self.state = .checkout(checkoutState)
-        self.basedOn = nil
-        self.subpath = subpath
-    }
+        internal init(
+            packageRef: PackageReference,
+            state: State,
+            subpath: RelativePath
+        ) {
+            self.packageRef = packageRef
+            self.subpath = subpath
+            self.state = state
+        }
 
-    /// Create a dependency present locally on the filesystem.
-    public static func local(
-        packageRef: PackageReference
-    ) -> ManagedDependency {
-        return ManagedDependency(
-            packageRef: packageRef,
-            state: .local,
-            // FIXME: This is just a fake entry, we should fix it.
-            subpath: RelativePath(packageRef.identity.description),
-            basedOn: nil
-        )
-    }
+        /// Create an editable managed dependency based on a dependency which
+        /// was *not* in edit state.
+        ///
+        /// - Parameters:
+        ///     - subpath: The subpath inside the editable directory.
+        ///     - unmanagedPath: A custom absolute path instead of the subpath.
+        public func edited(subpath: RelativePath, unmanagedPath: AbsolutePath?) -> ManagedDependency {
+            return .edited(packageRef: self.packageRef, subpath: subpath, basedOn: self, unmanagedPath: unmanagedPath)
+        }
 
-    internal init(
-        packageRef: PackageReference,
-        state: State,
-        subpath: RelativePath,
-        basedOn: ManagedDependency?
-    ) {
-        self.packageRef = packageRef
-        self.subpath = subpath
-        self.basedOn = basedOn
-        self.state = state
-    }
+        /// Create a dependency present locally on the filesystem.
+        public static func local(
+            packageRef: PackageReference
+        ) -> ManagedDependency {
+            return ManagedDependency(
+                packageRef: packageRef,
+                state: .local,
+                // FIXME: This is just a fake entry, we should fix it.
```

Comment:
Is that for a later PR, or a FIXME for a future update of this one?

Reply:
this FIXME was there before, so not a new thing

Reply:
I do think we should fix this, but it was not in scope for this PR